daemon/errors.go

```diff
@@ -132,35 +132,46 @@ func (e startInvalidConfigError) Error() string {
 
 func (e startInvalidConfigError) InvalidParameter() {} // Is this right???
 
-func translateContainerdStartErr(cmd string, setExitCode func(int), err error) error {
+// translateContainerdStartErr converts the error returned by containerd (if any)
+// when starting a container, and returns the corresponding exit-code for the
+// container and an errdefs error (either errdefs.ErrInvalidParameter or errdefs.ErrUnknown).
+//
+// exitCode can be one of:
+// - -1:   unknown error (do not update the container's exitcode)
```

Comment:
FWIW; I was a bit in doubt what the most logical value would be for "we don't have a new exit code for you"; we could use `0` (default for `int`), but possibly this would be confusing (as `0` means "no error"); So, I landed on `-1`, which is not a valid exit-code, and as indicator "we don't know, so keep it as-is"

Perhaps it could be changed to `1` (there was an error!) But this would potentially be overwriting the actual container's exit code? (not sure if it's set already by this time though)